Sources/MongoSwift/Cursor.swift

```diff
@@ -34,24 +34,50 @@ public class MongoCursor: Sequence, IteratorProtocol {
     }
 
     /**
-     * Returns the next document in this cursor, or nil
+     * Returns the next document in this cursor, or nil. Throws an error if one
+     * occurs. (Compared to next(), which returns nil and requires manually checking
+     * for an error afterward.)
+     *
      */
-    public func next() -> Document? {
+    func nextOrError() throws -> Document? {
         let out = UnsafeMutablePointer<UnsafePointer<bson_t>?>.allocate(capacity: 1)
         defer {
             out.deinitialize(count: 1)
             out.deallocate(capacity: 1)
         }
-        var error = bson_error_t()
 
         if !mongoc_cursor_next(self._cursor, out) {
+            var error = bson_error_t()
             if mongoc_cursor_error(self._cursor, &error) {
-                print("cursor error: (domain: \(error.domain), code: \(error.code), message: \(toErrorString(error)))")
+                throw MongoError.invalidCursor(message: toErrorString(error))
             }
-
             return nil
         }
+        return Document(fromPointer: UnsafeMutablePointer(mutating: out.pointee!))
+    }
 
+    /**
+     *  Returns the error that occurred while iterating this cursor, if one exists. 
+     *  This function should be called to check for errors after next() returns nil. 
+     */
+    func getError() -> Error? {
```

Comment:
is it more idiomatic to have a computed variable rather than a getter here? `getError` feels very Java to me

Reply:
I agree, sgtm